cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunExecutableTask.java

```diff
@@ -64,71 +65,59 @@
-                                "'.");
-                    case SINGLE_MODULE:
-                        SingleModuleContext singleModuleContext = buildContext.get(BuildContextField.SOURCE_CONTEXT);
-                        throw createLauncherException("no entry points found in '" +
-                                singleModuleContext.getModuleName() + "'.");
-                    default:
-                        throw createLauncherException("unknown source type found when running executable.");
-                }
+        // check if any entry points are found.
+        if (executableModule == null) {
+            switch (buildContext.getSourceType()) {
+                case SINGLE_BAL_FILE:
+                    SingleFileContext singleFileContext = buildContext.get(BuildContextField.SOURCE_CONTEXT);
+                    throw createLauncherException(String.format("no entry points found in '%s'.",
+                            singleFileContext.getBalFile()));
+                case SINGLE_MODULE:
+                    SingleModuleContext singleModuleContext = buildContext.get(BuildContextField.SOURCE_CONTEXT);
+                    throw createLauncherException(String.format("no entry points found in '%s'.",
+                            singleModuleContext.getModuleName()));
+                default:
+                    throw createLauncherException("unknown source type found when running executable.");
             }
         }
 
-        if (!this.executablePath.isAbsolute()) {
-            this.executablePath = sourceRootPath.resolve(executablePath);
+        // if the executable does not exist.
+        if (Files.notExists(this.executableJarPath)) {
+            throw createLauncherException(String.format("cannot run '%s' as it does not exist.",
+                    executableJarPath.toAbsolutePath().toString()));
         }
 
-        // clean up the path and get absolute path
-        this.executablePath = this.executablePath.toAbsolutePath().normalize();
-
-        // if the executable does not exist
-        if (Files.notExists(this.executablePath)) {
-            throw createLauncherException("cannot run '" + this.executablePath.toAbsolutePath().toString() +
-                    "' as it does not exist.");
-        }
-
-        // if the executable is not a file and not an extension with .jar
-        if (!(Files.isRegularFile(this.executablePath) &&
-                this.executablePath.toString().endsWith(BLANG_COMPILED_JAR_EXT))) {
-
-            throw createLauncherException("cannot run '" + this.executablePath.toAbsolutePath().toString() +
-                    "' as it is not an executable with .jar extension.");
+        // if the executable is not a file and not an extension with .jar.
+        if (!(Files.isRegularFile(this.executableJarPath) &&
+                this.executableJarPath.toString().endsWith(BLANG_COMPILED_JAR_EXT))) {
```

Comment:
Do we need this check. Executable always jar file

Reply:
I added the same check we had for the final uber jar earlier. If we dont need this for the module jar, we can remove.

Reply:
Removing the check as per the offline discussion.